internal/graph/cycle_check_resolver_test.go

```diff
@@ -0,0 +1,81 @@
+package graph
+
+import (
+	"context"
+	"testing"
+
+	"github.com/oklog/ulid/v2"
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
+
+	"github.com/openfga/openfga/pkg/storage"
+	"github.com/openfga/openfga/pkg/storage/memory"
+	"github.com/openfga/openfga/pkg/testutils"
+	"github.com/openfga/openfga/pkg/tuple"
+	"github.com/openfga/openfga/pkg/typesystem"
+)
+
+func TestCycleDetectionCheckResolver(t *testing.T) {
+	ds := memory.New()
+	storeID := ulid.Make().String()
+
+	err := ds.Write(context.Background(), storeID, nil, []*openfgav1.TupleKey{
+		tuple.NewTupleKey("document:1", "viewer3", "document:1#viewer3"),
+	})
+	require.NoError(t, err)
+
+	model := testutils.MustTransformDSLToProtoWithID(`model
+	schema 1.1
+  
+  type user
+  
+  type document
+	relations
+		define viewer1: [user, document#viewer1]
+		define viewer2: viewer1 or viewer2
+		define viewer3: viewer1 or viewer2`)
+
+	ctx := typesystem.ContextWithTypesystem(context.Background(), typesystem.New(model))
+
+	ctx = storage.ContextWithRelationshipTupleReader(ctx, ds)
+
+	cycleDetectionCheckResolver := NewCycleDetectionCheckResolver()
+	localCheckResolver := NewLocalChecker()
+
+	t.Cleanup(func() {
+		cycleDetectionCheckResolver.Close()
+		localCheckResolver.Close()
+		goleak.VerifyNone(t)
+	})
```

Comment:
please put each of these cleanups right after the resource has been created.

```go
func TestCycleDetectionCheckResolver(t *testing.T) {
t.Cleanup(goleak.VerifyNone(t))

cycleDetectionCheckResolver = NewCycleDetectionCheckResolver()
t.Cleanup(cycleDetectionCheckResolver.Close)

localCheckResolver = NewLocalChecker()
t.Cleanup(localCheckResolver.Close)
```

Reply:
Ok. But seeing as this is stylistic, can you please encode this as a linting rule? We have no consistency in the codebase currently.